File: JAVA/level2/타겟_넘버.java
class Solution {
    int answer;
    int result;
    int SIZE;

    public void calculate(int index, int val, int[] numbers){
        if(index == SIZE){
            if(val == result)
                answer++;
        }
        else{
            calculate(index+1, val+numbers[index], numbers);
            calculate(index+1, val-numbers[index], numbers);
        }
    }

    public int solution(int[] numbers, int target) {
        answer = 0;
        result = target;
        SIZE = numbers.length;

        calculate(0, 0, numbers);

        return answer;
    }
}

File: JavaScript/level2/타겟_넘버.js
function check(numbers, target, index, val){
    if(index === numbers.length){
        if(val === target)
           return 1;
        return 0 ;
    }
    
    return(check(numbers, target, index+1, val+numbers[index]) + check(numbers, target, index+1, val-numbers[index]));  
}

function solution(numbers, target) {
    
    return check(numbers, target, 0, 0);
}